optuna/storages/_rdb/storage.py

```diff
@@ -981,6 +981,8 @@ def record_heartbeat(self, trial_id: int) -> None:
                 heartbeat = models.TrialHeartbeatModel(trial_id=trial_id)
                 session.add(heartbeat)
             else:
+                heartbeat = models.TrialHeartbeatModel.where_trial_id(trial_id, session, True)
```

Comment:
Why do we need to re-define the object here?

Reply:
In line 979, `models.TrialHeartbeatModel.where_trial_id` will not write to the DB because the third argument is omitted, and return None if it does not exist.

In Line 984, we have confirmed that the object exists, so we set the third argument to `True` and issue a query with write.